src/lx/parser.act

```diff
@@ -350,17 +370,20 @@
 
 	<compile-regex>: (s :string, f :flags) -> (r :fsm) = @{
 		struct re_err err;
-		const char *s;
+		struct re_buf buf;
```

Comment:
Would it work out neater to use `struct re_but` for the `:string` type? I'm not sure.